tutorials/kotlin-springboot-app-engine.md

```diff
@@ -0,0 +1,214 @@
+Google's products. Using [Kotlin](https://kotlinlang.org/) and [Spring Boot](https://projects.spring.io/spring-boot/), in this tutorial you'll
+see how to deploy your application to App Engine. 
+
+
+You will create a new Spring Boot application, and then you will learn how to:
+
+*   Deploy your application
+*   Update your application
+
+While the tutorial uses Kotlin 1.2 and Spring Boot 2 M7, other releases of Kotlin and Spring Boot should work
+without any modifications (other than version numbers in Maven files). This tutorial does assume you're familiar 
+with Spring Boot and creating web applications. For simplicity the tutorial responds with JSON to a specific HTTP request, but can
+be built-on to connect to other Google services and/or databases.
+
+## Before you begin
+
+Before running this tutorial, you must set up a Google Cloud Platform project,
+and you need to have Docker and the Google Cloud SDK installed.
+
+Create a project that will host your Spring Boot application. You can also reuse
+an existing project.
+
+1.  Use the [Google Cloud Platform Console](https://console.cloud.google.com/)
+    to create a new Cloud Platform project. Remember the project ID; you will
+    need it later. Later commands in this tutorial will use `${PROJECT_ID}` as
+    a substitution, so you might consider setting the `PROJECT_ID` environment
+    variable in your shell.
+
+2.  Enable billing for your project.
+
+
+3.  Install the [Google Cloud SDK](https://cloud.google.com/sdk/). Make sure
+    you [initialize](https://cloud.google.com/sdk/docs/initializing) the SDK
+    and set the default project to the new project you created.
+
+    Version 175.0.0 or later of the SDK is required.
+
+3.  Install [JDK 8 or higher](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) if you do not already have it. 
+
+4.  Install [Maven](https://maven.apache.org/install.html).
+
+## Creating a new app and running it locally
+
+In this section, you will create a new Spring Boot app and make sure it runs. If
+you already have an app to deploy, you can use it instead.
+
+1. Use [start.spring.io](https://start.spring.io) to generate a Spring Boot application using Kotlin as the language, Maven as the build system. Alternatively,
+you can [download](https://github.com/jetbrains/gcp-samples) the sample application. 
+
+2. Download the generated project and save it to a local folder.
+
+3. Open the resulting project in your favourite IDE or editor and create a new source file named `MessageController.kt` with the following contents:
+
+```kotlin
+package com.jetbrains.demo
+
+import org.springframework.web.bind.annotation.*
+
+data class Message(val text: String, val priority: String)
+
+@RestController
+class MessageController {
+    @RequestMapping("/message")
+    fun message(): Message {
+        return Message("Hello from Google Cloud", "High")
+    }
+}
+```
+
+The package should match that of your group and artifact name. 
```

Comment:
The file contents and following sentence are not getting indented as part of step 3. Could you try ```shell to see if that works better. Or you could indent 8 spaces.

Reply:
Do you mean the Kotlin code? It's using three ticks with Kotlin...